### PR TITLE
[telemetry] Add telemetry to SbPlayerWriteSamples

### PR DESCRIPTION
**Pull request by borongc**
Adding additional Media Telemetry to cover SbPlayerWriteSamples.

The rough range of each function:
1. When batched write is enabled (Media.MaxAudioSamplesPerWrite=256, https://github.com/youtube/cobalt/pull/3271):
    1. SbPlayerWriteSamples()/audio: 1us to 98us
    2. SbPlayerWriteSamples()/video: 1us to 3.4ms
2. When batched write is disabled (Media.MaxAudioSamplesPerWrite=1, https://github.com/youtube/cobalt/pull/3271):
    1. SbPlayerWriteSamples()/audio: 9us to 65.47ms
    2. SbPlayerWriteSamples()/video: 8us to 28.07ms

b/303328238

### CODE REVIEW

File: cobalt/media/base/metrics_provider.cc [hunk 1]
```diff
@@ -122,6 +122,17 @@ void MediaMetricsProvider::EndTrackingAction(MediaAction action) {
   tracked_actions_start_times_.erase(action);
 }
 
+void MediaMetricsProvider::EndTrackingAction(MediaAction action,
+                                             int number_of_samples) {
+  DCHECK(IsActionCurrentlyTracked(action));
+  base::AutoLock scoped_lock(mutex_);
+
+  auto duration = clock_->NowTicks() - tracked_actions_start_times_[action];
+  auto duration_per_sample = duration / number_of_samples;
```
Comment (at-ninja) [inline cobalt/media/base/metrics_provider.cc:131]: Should `number_of_samples` be asserted to be non-zero?

File: tools/metrics/histograms/metadata/cobalt/histograms.xml [hunk 2]
```diff
@@ -338,6 +338,24 @@ Always run the pretty print utility on this file after editing:
   <summary>Timing data for the destruction of SbPlayer.</summary>
 </histogram>
 
+<histogram name="Cobalt.Media.SbPlayer.WriteSamples.Audio.LatencyTiming"
+    units="microseconds" expires_after="never">
+<!-- expires-never: Needed for long-term tracking of SbPlayer latencies. -->
+
+  <owner>borongchen@google.com</owner>
+  <owner>cobalt-team@google.com</owner>
+  <summary>Timing data for writing audio samples.</summary>
```
Comment (at-ninja) [inline tools/metrics/histograms/metadata/cobalt/histograms.xml:347]: I think these should mention that they are per-sample measurements, and not the entire sampled time.

Comment (borongc) [reply tools/metrics/histograms/metadata/cobalt/histograms.xml:347]: Fixed.

File: cobalt/media/base/metrics_provider.cc (hunk 1 above)
Comment (borongc) [reply cobalt/media/base/metrics_provider.cc]: Fixed.

**Review by at-ninja — APPROVED** [review]

**Review by joeltine — APPROVED** [review]

**Pull request merged**